Tests/Add-ZipArchiveEntry.Tests.ps1

```diff
@@ -0,0 +1,273 @@
+                if( $ExpectedContent )
+                {
+                    $reader = New-Object 'IO.StreamReader' ($entry.Open())
+                    try
+                    {
+                        $content = $reader.ReadToEnd()
+                        $content | Should -Be $ExpectedContent
+                    }
+                    finally
+                    {
+                        $reader.Close()
+                    }
+                }
+            }
+        }
+    }
+    finally
+    {
+        $file.Dispose()
+    }
+}
+
+function ThenArchiveEmpty
+{
+    [IO.Compression.ZipArchive]$file = [IO.Compression.ZipFile]::OpenRead($archive.FullName)
+    try
+    {
+        It ('should not add any files to ZIP') {
+            $file.Entries | Should -BeNullOrEmpty
+        }
+    }
+    finally
+    {
+        $file.Dispose()
+    }    
+}
+
+function ThenArchiveNotContains
+{
+    param(
+        [string[]]
+        $Entry
+    )
+
+    [IO.Compression.ZipArchive]$file = [IO.Compression.ZipFile]::OpenRead($archive.FullName)
+    try
+    {
+        It ('should not add files to ZIP') {
+            foreach( $entryName in $Entry )
+            {
+                $file.GetEntry($entryName) | Should -BeNullOrEmpty
+            }
+        }
+    }
+    finally
+    {
+        $file.Dispose()
+    }
+}
+
+function ThenError
+{
+    param(
+        $Matches
+    )
+
+    It ('should write an error') {
+        $Global:Error | Should -Match $Matches
+    }
+}
+
+function  WhenAddingFiles
+{
+    [CmdletBinding()]
+    param(
+        [string[]]
+        $Path,
+
+        [Switch]
+        $Force,
+
+        $AtArchiveRoot,
+
+        [Switch]
+        $NoPipeline,
+
+        [Switch]
+        $NoBasePath,
+
+        [string]
+        $AtBasePath
+    )
+
+    $archivePath = Join-Path -Path $TestDrive.FullName -ChildPath 'zip.zip'
+    if( -not (Test-Path -Path $archivePath -PathType Leaf) )
+    {
+        $script:archive = New-ZipArchive -Path $archivePath
+    }
+
+    $params = @{
+                    ZipArchivePath = $archive.FullName;
+                }
+    if( -not $NoBasePath )
+    {
+        $params['BasePath'] = $TestDrive.FullName;
+    }
+
+    if( $AtBasePath )
+    {
+        $params['BasePath'] = $AtBasePath
+    }
+
+    if( $AtArchiveRoot )
+    {
+        $params['EntryParentPath'] = $AtArchiveRoot
+    }
+
+    if( $Force )
+    {
+        $params['Force'] = $true
+    }
+
+    $Global:Error.Clear()
+    if( $NoPipeline )
+    {
+        Push-Location -Path $TestDrive.FullName
+        try
+        {
+            foreach( $item in $Path )
+            {
+                Add-ZipArchiveEntry @params -InputObject $item
+            }
+        }
+        finally
+        {
+            Pop-Location
+        }
+    }
+    else
+    {
+        Find-GlobFile -Path $TestDrive.FullName -Include $Path | Add-ZipArchiveEntry @params
+    }
+}
+
+Describe 'Add-ZipArchiveEntry' {

```

Comment:
Additional test: What happens when `Add-ZipArchiveEntry` is passed a path that **does not** exist under the `BasePath` directory?
```
'C:\dir\file.txt', 'C:\some\other\dir\second.txt' | Add-ZipArchiveEntry -ZipArchivePath 'archive.zip' -BasePath 'C:\dir'
```